### PR TITLE
Send logs to socket (in ModHelper project)

### PR DESCRIPTION
**Pull request by Raicuparta**

### CODE REVIEW

File: createrelease.bat [hunk 1]
```diff
@@ -23,6 +23,7 @@ copy "OWML.Nuget\bin\Debug\OWML.ModHelper.Events.dll" "Release\OWML.ModHelper.Ev
 copy "OWML.Nuget\bin\Debug\OWML.ModHelper.Assets.dll" "Release\OWML.ModHelper.Assets.dll"
 copy "OWML.Nuget\bin\Debug\OWML.ModHelper.Interaction.dll" "Release\OWML.ModHelper.Interaction.dll"
 copy "OWML.Nuget\bin\Debug\OWML.ModHelper.Menus.dll" "Release\OWML.ModHelper.Menus.dll"
+copy "OWML.Nuget\bin\Debug\OWML.ModHelper.Logging.dll" "Release\OWML.ModHelper.Logging.dll"
```
Comment (amazingalek) [inline createrelease.bat:26]: Nope? :)

File: OWML.ModHelper/ModConsole.cs [hunk 2]
```diff
@@ -1,56 +1,37 @@
 ﻿using System;
-using System.IO;
-using System.Linq;
-using System.Text;
 using OWML.Common;
 
 namespace OWML.ModHelper
 {
-    public class ModConsole : IModConsole
+    public abstract class ModConsole : IModConsole
```
Comment (amazingalek) [inline OWML.ModHelper/ModConsole.cs:6]: ModOutput?

File: OWML.ModHelper/ModConsole.cs [hunk 3]
```diff
@@ -1,56 +1,37 @@
...
+        protected readonly IModLogger _logger;
+        protected readonly IModManifest _manifest;
+        protected readonly IOwmlConfig _owmlConfig;
 
-        private readonly IModLogger _logger;
-        private readonly IModManifest _manifest;
+        public abstract void WriteLine(string s);
+        public abstract void WriteLine(params object[] s);
 
         public ModConsole(IOwmlConfig config, IModLogger logger, IModManifest manifest)
```
Comment (amazingalek) [inline OWML.ModHelper/ModConsole.cs:20]: protected

File: OWML.Patcher/OWPatcher.cs [hunk 4]
```diff
@@ -31,10 +31,21 @@ public void PatchGame()
...
-                "Newtonsoft.Json.dll", "System.Runtime.Serialization.dll", "0Harmony.dll", "NAudio-Unity.dll",
-                "OWML.Manifest.json", "OWML.ModHelper.Interaction.dll" };
+            var filesToCopy = new[] {
+                "OWML.ModLoader.dll",
+                "OWML.Common.dll",
+                "OWML.ModHelper.dll",
+                "OWML.ModHelper.Events.dll",
+                "OWML.ModHelper.Assets.dll",
+                "OWML.ModHelper.Menus.dll",
+                "OWML.ModHelper.Logging.dll",
```
Comment (amazingalek) [inline OWML.Patcher/OWPatcher.cs:41]: Nope

File: OWML.ModHelper/ModConsole.cs [hunk 5]
```diff
@@ -1,56 +1,37 @@
...
         [Obsolete("Use ModHelper.Console instead")]
         public static ModConsole Instance { get; private set; }
 
-        public static event Action<IModManifest, string> OnConsole; 
+        public static event Action<IModManifest, string> OnConsole;
 
-        private static FileStream _writer;
+        protected readonly IModLogger _logger;
+        protected readonly IModManifest _manifest;
+        protected readonly IOwmlConfig _owmlConfig;
```
Comment (amazingalek) [inline OWML.ModHelper/ModConsole.cs:15]: protected variables should start with capital letter. Logger etc.

**Comment by amazingalek** [comment]
Nice. I tested this with Mod Manager (aicuparta/owml-logs), works well :)
Had to remove the trace of `OWML.ModHelper.Logging.dll` to make it work.
Could you make a folder called Logging for all the logging/console/output files?

File: OWML.ModHelper/ModConsole.cs (hunk 2 above)
Comment (Raicuparta) [reply OWML.ModHelper/ModConsole.cs:6]: Needs to remain as ModConsole to avoid breaking backwards compatibility.

Comment (Raicuparta) [reply OWML.ModHelper/ModConsole.cs:6]: @amazingalek ^ (never sure how email notifications work here)

Comment (amazingalek) [reply OWML.ModHelper/ModConsole.cs]: Ah, because of the static event, right.

**Comment by amazingalek** [comment]
Perfect. The only thing missing is bumping the version top 0.3.49 :)

**Pull request merged**